### ISSUE TITLE
Why there is a need to clip gradient?

### ISSUE THREAD

**Issue by bowenc0221** [open-mmlab/mmdetection#107]
I notice there is an `optimizer_config` with `grad_clip=dict(max_norm=35, norm_type=2)` which is not done in either Detectron in Caffe2 or the PyTorch version.  
Will the training be unstable or even unable to converge without gradient clip?  
And why it is set to 35? Is it related to the batchsize?  

Thanks.

**Comment by jhyume** [comment]
Same question here.

**Comment by hellock** [comment]
Sorry for the late reply. Gradient clipping is used when we build and train our own models, which may diverge with large learning rates, and 35 is just a empirical value.
For classic frameworks like Faster R-CNN and Mask R-CNN, this is unnecessary. The clipping usually takes no effect and you can explicitly disable it by setting `grad_clip=None`.

**Comment by analyticsbot** [comment]
Does this value need to be higher for distributed training? I run into the loss=nan when this value is set to default of 35 when running with 4 GPU machine. However, with just 1 GPU in action, it works fine.
